Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -476,6 +483,22 @@ extension BookmarksPanel: HomePanelContextMenu {
     }
 }
 
+@available(iOS 11.0, *)
+extension BookmarksPanel: UITableViewDragDelegate {
+    func tableView(_ tableView: UITableView, itemsForBeginning session: UIDragSession, at indexPath: IndexPath) -> [UIDragItem] {
+        guard let bookmark = source?.current[indexPath.row] as? BookmarkItem, let url = URL(string: bookmark.url), let itemProvider = NSItemProvider(contentsOf: url) else {
```

Comment:
- The `HomePanelContextMenu` protocol already has a `getSiteDetails(for: indexPath)` which you can use to fetch the site from every panel in a generic way. 
- If `itemsForBeginning` is implemented on the `SiteTableViewController` all you need to do is 
`let panelVC = self as? HomePanelContextMenu, let site = panelVC.getSiteDetails(for: indexPath)`